pkg/metastore/compaction/compactor/metrics.go

```diff
@@ -66,3 +66,79 @@ func (b *queueStatsCollector) Collect(m chan<- prometheus.Metric) {
 	m <- prometheus.MustNewConstMetric(b.rejected, prometheus.CounterValue, float64(b.stats.rejected.Load()))
 	m <- prometheus.MustNewConstMetric(b.missed, prometheus.CounterValue, float64(b.stats.missed.Load()))
 }
+
+type globalQueueStatsCollector struct {
+	compactionQueue *compactionQueue
+
+	blocks      *prometheus.Desc
+	queues      *prometheus.Desc
+	backlogJobs *prometheus.Desc
+}
+
+const globalQueueMetricsPrefix = "compaction_global_queue_"
+
+func newBlockQueueStatsCollector(compactionQueue *compactionQueue) *globalQueueStatsCollector {
+	variableLabels := []string{"level"}
+
+	return &globalQueueStatsCollector{
+		compactionQueue: compactionQueue,
+
+		blocks: prometheus.NewDesc(
+			globalQueueMetricsPrefix+"blocks_current",
+			"The current total number of blocks across all queues.",
+			variableLabels, nil,
+		),
+
+		queues: prometheus.NewDesc(
+			globalQueueMetricsPrefix+"queues_current",
+			"The current total number of queues.",
+			variableLabels, nil,
+		),
+
+		backlogJobs: prometheus.NewDesc(
+			globalQueueMetricsPrefix+"backlog_jobs_current",
+			"The current estimated number of compaction jobs that are yet to be created.",
+			variableLabels, nil,
+		),
+	}
+}
+
+func (c *globalQueueStatsCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- c.blocks
+	ch <- c.queues
+	ch <- c.backlogJobs
+}
+
+func (c *globalQueueStatsCollector) Collect(ch chan<- prometheus.Metric) {
+	numLevels := len(c.compactionQueue.config.Levels)
+	for levelIdx := 0; levelIdx < numLevels; levelIdx++ {
+		var blocksForLevel int32
+		var queuesForLevel int
+		var backlogJobsForLevel int
+
+		levelLabel := strconv.Itoa(levelIdx)
+
+		if levelIdx < len(c.compactionQueue.levels) && c.compactionQueue.levels[levelIdx] != nil {
+			queue := c.compactionQueue.levels[levelIdx]
+
+			maxBlocks := queue.config.maxBlocks(uint32(levelIdx))
+			if maxBlocks == 0 {
+				// This is likely a misconfiguration, we'll just skip it.
+				continue
+			}
+
+			for _, staged := range queue.staged {
```

Comment:
There are race conditions: access to `c.compactionQueue.levels` and ranging sub-queues via `queue.staged`.

I recommend counting items as they are added and removed: add a global counterpart of the `queueStats` struct and update it together with the fine-grained stat counters (4 places).

To track the number of queues per level, we want to catch the moment when `staged` objects are created and deleted (2 places).